src/box/alter.cc

```diff
@@ -4418,9 +4418,27 @@ on_replace_dd_schema(struct trigger * /* trigger */, void *event)
 
 /** Unregister the replica affected by the change. */
 static int
-on_replace_cluster_clear_id(struct trigger *trigger, void * /* event */)
-{
-	replica_clear_id((struct replica *)trigger->data);
+on_replace_cluster_clear_id(struct trigger *trigger, void *event)
+{
+	struct replica *replica = (struct replica *)trigger->data;
+	struct txn_stmt *stmt = (struct txn_stmt *)event;
+	(void)stmt;
+	if (replica->id == instance_id) {
+		assert(stmt->row->replica_id != 0 ||
+		       recovery_state != FINISHED_RECOVERY);
+		if (recovery_state == FINISHED_RECOVERY) {
+			diag_set(ClientError, ER_LOCAL_INSTANCE_ID_IS_READ_ONLY,
+				 (unsigned)instance_id);
+			struct diag *diag = diag_get();
+			replicaset_foreach(replica) {
+				if (replica->applier != NULL)
+					applier_kill(replica->applier,
+						     diag_last_error(diag));
+			}
+			diag_clear(diag);
+		}
+	}
```

Comment:
Don't we also wanna stop the relay on the over nodes to the deleted replica? So that it becomes orphan and we can delete it in `replica_clear_id`?

Reply:
I'd say we want to stop relays **on** the deleted node to the others.
Because basically it has become "anonymous", and we don't allow to replicate from anonymous nodes.

The relays on the other nodes to this replica will be stopped just because the replica stops its appliers.

Reply:
Added relay stopping in the scope of #10266.

Reply:
Resolving since the PR is approved.